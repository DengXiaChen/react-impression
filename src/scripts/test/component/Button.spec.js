import React from 'react';
import { shallow } from 'enzyme';
import { Button } from '../../components/impression';

const setup = () => {
    const props = {
        theme: 'primary',
        onClick: jest.fn(),
        outline: false,
        size: 'lg',
        shape: 'pill',
        block: true,
        disabled: true,
    };

    const wrapper = shallow(<Button {...props} />);

    return {
        props,
        wrapper,
    };
};

describe('Button', () => {
    const { props, wrapper } = setup();

    // 通过 input 是否存在来判断 Todo组件是否被渲染
    it('render', () => {
        expect(wrapper.find('button').exists()).toBe(true);
    });

    // 当点击 单选按钮，onClick 方法应该被调用
    it('click', () => {
        wrapper.find('button').simulate('click');
        expect(props.onClick).toHaveBeenCalled();
    });

    it('theme', () => {
        expect(wrapper.find('.btn-primary').exists()).toBe(true);
    });

    it('size', () => {
        expect(wrapper.find('.btn-lg').exists()).toBe(true);
    });

    it('shape', () => {
        expect(wrapper.find('.btn-pill').exists()).toBe(true);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
