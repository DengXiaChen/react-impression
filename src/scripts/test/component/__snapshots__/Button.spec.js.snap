// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="btn btn-primary btn-lg btn-pill btn-block"
    disabled={true}
    href={undefined}
    onClick={[Function]}
    type="button"
/>,
  "nodes": Array [
    <button
      className="btn btn-primary btn-lg btn-pill btn-block"
      disabled={true}
      href={undefined}
      onClick={[Function]}
      type="button"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Button
        block={true}
        disabled={true}
        onClick={[Function]}
        outline={false}
        shape="pill"
        size="lg"
        theme="primary"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "block": true,
          "disabled": true,
          "onClick": [Function],
          "outline": false,
          "shape": "pill",
          "size": "lg",
          "theme": "primary",
        },
        "refs": Object {},
        "state": Object {
          "Tag": "button",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="btn btn-primary btn-lg btn-pill btn-block"
          disabled={true}
          href={undefined}
          onClick={[Function]}
          type="button"
/>,
        "_debugID": 2,
        "_renderedOutput": <button
          className="btn btn-primary btn-lg btn-pill btn-block"
          disabled={true}
          href={undefined}
          onClick={[Function]}
          type="button"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    block={true}
    disabled={true}
    onClick={[Function]}
    outline={false}
    shape="pill"
    size="lg"
    theme="primary"
/>,
}
`;
